internal/dot/graph.go

```diff
@@ -47,24 +47,39 @@ type Ctor struct {
 	Params      []*Param
 	GroupParams []*Group
 	Results     []*Result
+	ResultCtor  map[nodeKey][]*Ctor
 	ErrorType   ErrorType
 }
 
+type nodeKey struct {
+	t     reflect.Type
+	name  string
+	group string
+}
```

Comment:
nodeKey is a superset of groupKey. You can drop groupKey if you're using nodeKey.

Reply:
nodeKey is used to for comparison purposes, groupKey is used to quickly generate at a key to lookup the group object in dg.groupMap and still has a purpose outside of nodeKey. If i removed groupKey I would have to manually construct the groupKey more than once.

Reply:
Discussed offline: It'll still work with just nodeKey. We can delete groupKey completely.

Reply:
I removed it and casted nodeKey into groupKey because it was required to lookup a node's equivalent value in the groupMap.